src/ansys/units/quantity_dimensions.py

```diff
@@ -0,0 +1,267 @@
+﻿# Copyright (C) 2023 - 2025 ANSYS, Inc. and/or its affiliates.
+
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+Defines immutable objects representing SI base and derived physical dimensions.
+
+This module provides globally defined `Dimensions` instances corresponding to the
+seven SI base quantities (e.g., mass, time) and many commonly used derived quantities
+(e.g., force, energy, pressure). These dimension definitions are used internally
+for dimensional analysis, quantity construction, and validation.
+
+Each dimension is represented as a `Dimensions` object, which captures the
+dimensional exponents of base quantities such as mass, length, and time.
+
+This module is not intended for direct modification at runtime. All objects
+are statically defined and immutable.
+
+Examples
+--------
+from ansys.units.dimensions import Dimensions
+from ansys.units.quantity_dimensions import QuantityDimensions
+
+force = QuantityDimensions.FORCE
+pressure = QuantityDimensions.PRESSURE
+pressure == force / QuantityDimensions.AREA
+True
```

Comment:
We follow Numpy doc style. Following style helps to identify difference between Python code and it's output. Same should be applied for the remaining cases, if there are any. 



```suggestion
Examples
--------
>>> from ansys.units.dimensions import Dimensions
>>> from ansys.units.quantity_dimensions import QuantityDimensions

>>> force = QuantityDimensions.FORCE
>>> pressure = QuantityDimensions.PRESSURE
>>> pressure == force / QuantityDimensions.AREA
True
```